costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -90,23 +90,27 @@ Costmap2DROS::Costmap2DROS(std::string name, tf::TransformListener& tf) :
   // get two frames
   private_nh.param("global_frame", global_frame_, std::string("/map"));
   private_nh.param("robot_base_frame", robot_base_frame_, std::string("base_link"));
+  private_nh.param("robot_ground_frame", robot_ground_frame_, global_frame_);
 
   // make sure that we set the frames appropriately based on the tf_prefix
   global_frame_ = tf::resolve(tf_prefix, global_frame_);
   robot_base_frame_ = tf::resolve(tf_prefix, robot_base_frame_);
+  robot_ground_frame_ = tf::resolve(tf_prefix, robot_ground_frame_);
 
   ros::Time last_error = ros::Time::now();
   std::string tf_error;
   // we need to make sure that the transform between the robot base frame and the global frame is available
   while (ros::ok()
-      && !tf_.waitForTransform(global_frame_, robot_base_frame_, ros::Time(), ros::Duration(0.1), ros::Duration(0.01),
-                               &tf_error))
+      && (!tf_.waitForTransform(global_frame_, robot_base_frame_, ros::Time(), ros::Duration(0.1), ros::Duration(0.01),
+                               &tf_error)
+      || !tf_.waitForTransform(robot_ground_frame_, robot_base_frame_, ros::Time(), ros::Duration(0.1), ros::Duration(0.01),
+                               &tf_error)))
```

Comment:
Why not directly check for global_frame_ and robot_ground_frame_?

Reply:
Yes, thanks, I mixed it up. Corrected it now on last committ.